src/Indicator.vala

```diff
@@ -18,51 +18,71 @@
  */
 
 public class Power.Indicator : Wingpanel.Indicator {
-    private bool is_in_session = false;
+    private const string DBUS_PATH = "/org/gnome/SettingsDaemon/Power";
+    private const string DBUS_NAME = "org.gnome.SettingsDaemon";
 
-    private Widgets.DisplayWidget? display_widget = null;
+    public bool is_in_session { get; construct set; }
 
+    private Widgets.DisplayWidget? display_widget = null;
     private Widgets.PopoverWidget? popover_widget = null;
 
     private Services.Device primary_battery;
+    private Services.DBusInterfaces.PowerSettings screen;
+
     private bool notify_battery = false;
 
+    private int screen_brightness {
```

Comment:
don't do this ugly way of getters and setters every again. if you need logic in your set functions then just make a normal set function. Also make the set of the property private.

Reply:
I do not have a problem with this - property functions are commonly used in elementary code (although less commonly for private properties).  This is neater and less error prone - using a separate setter function means that `screen_brightness` could be set directly without range checking.

Reply:
Guys, please. I just wanted to add a small feature... 😄